Disable this test due to regression of Clang 20,
should be removed when https://github.com/catchorg/Catch2/issues/2991 is fixed.
--- a/tests/SelfTest/UsageTests/Misc.tests.cpp.orig	2025-12-29 05:31:54 +0800
+++ b/tests/SelfTest/UsageTests/Misc.tests.cpp	2026-01-11 18:23:00 +0800
@@ -388,9 +388,9 @@
     REQUIRE(x.size() > 0);
 }
 
-TEMPLATE_PRODUCT_TEST_CASE("Product with differing arities", "[template][product]", std::tuple, (int, (int, double), (int, double, float))) {
-    REQUIRE(std::tuple_size<TestType>::value >= 1);
-}
+// TEMPLATE_PRODUCT_TEST_CASE("Product with differing arities", "[template][product]", std::tuple, (int, (int, double), (int, double, float))) {
+//     REQUIRE(std::tuple_size<TestType>::value >= 1);
+// }
 
 using MyTypes = std::tuple<int, char, float>;
 TEMPLATE_LIST_TEST_CASE("Template test case with test types specified inside std::tuple", "[template][list]", MyTypes)
